Kernel/waitQueueADT.c

```diff
@@ -91,6 +91,7 @@ int wq_unblockSingle(TWaitQueue queue) {
         queue->offset = (queue->offset + 1) % queue->bufSize;
         queue->count--;
 
+        //Check if there is a chance of leaving the cpu to the process that was waiting for a semaphore
```

Comment:
This would not be handled by the `waitQueueADT` (since it's generic, not semaphore-specific). The semaphore system can decide to call `sch_yieldProcess()` after a `sch_blockProcess()`